Document hybrid apps with SDK v2 (#576)
DataDog/dd-sdk-reactnative: docs/hybrid_applications.md

# Hybrid applications

> Note: This document uses SDK v2. For v1 documentation, refer to the [previous version of this document][10].

Reviewer: ```suggestion
Note: This document uses SDK v2. For v1 documentation, see the [previous version of this document][10].
```

    - Through _auto-instrumentation_ - Some React classes and methods are modified to automate this.
    - Through _manual instrumentation_ - For example, if you want to report something you consider an error but that is not going to crash the app.
      Auto-instrumentation for JavaScript errors, resources, and interactions can only be started from JavaScript code.

> 2. You can share the same instance of the core SDK between native and React Native. That way once you initialize the native SDK on the native side or on the React Native side (by calling `DdSdkReactNative.initialize`) it is initialized for both and events will appear in the same RUM Session. React Native uses the default core instance.

Reviewer: ```suggestion
2. You can share the same instance of the core SDK between native and React Native. This allows you to initialize the native SDK on either the native side or on the React Native side (by calling `DdSdkReactNative.initialize`) and have it initialized for both sides, with events appearing in the same RUM session. React Native uses the default core instance.
```

If you are sure that you don't call the native SDK before the React Native SDK this is the solution we recommend.

On Android, add the Datadog Android SDKs to your dependencies in your `android/app/build.gradle` file:

```groovy
    // The version will be set by @datadog/mobile-react-native
    implementation "com.datadoghq:dd-sdk-android-rum"
    implementation "com.datadoghq:dd-sdk-android-logs"
    implementation "com.datadoghq:dd-sdk-android-trace"
    implementation "com.datadoghq:dd-sdk-android-webview"
```

> On iOS, add the Datadog iOS SDKs to your dependencies in your `ios/Podfile` to use it in Objective C files:

Reviewer: ```suggestion
On iOS, add the Datadog iOS SDKs to your dependencies in your `ios/Podfile` to use in Objective C files:
```
